tests/unittests/Foundation/NSSetTests.mm

```diff
@@ -60,4 +60,24 @@
     }];
 
     ASSERT_OBJCEQ(expectedEvensLessThanFive, actual);
+}
+
+TEST(NSSet, FilterUsingPredicateString) {
+    NSSet* americanMakes = [NSSet setWithObjects:@"Chrysler", @"Ford", @"General Motors", nil];
+    NSPredicate* predicate = [NSPredicate predicateWithFormat:@"SELF BEGINSWITH[c] %@", @"G"];
+
+    NSSet* filteredSet = [americanMakes filteredSetUsingPredicate:predicate];
+
+    ASSERT_NE(nil, filteredSet);
+    ASSERT_EQ(1, [filteredSet count]);
+}
+
+TEST(NSSet, FilterUsingPredicateNumbers) {
+    NSSet* americanMakes = [NSSet setWithObjects:@2, @4, @8, @16, @32, nil];
+    NSPredicate* predicate = [NSPredicate predicateWithFormat:@"(SELF >= 4) AND (SELF < 32)"];
```

Comment:
>eWithFormat:@"(SELF >= 4) AND (SELF < 32)"]; [](start = 50, length = 44)

please test with an always-false predicate (removes all objects) and an always-true predicate.